### DISCUSSION TITLE
module not found dotenv

### DISCUSSION THREAD

**Discussion by GhouI** [vercel/community#2619] [Help] [answered]
### Summary

I am currently attempting to deploy the backend API for my server, but I keep encountering a "module not found" error.
My Code:
```javascript
const express = require('express');
const dotenv = require('dotenv');
const mysql = require('mysql');
const app = express();
const cors = require('cors');

const port = process.env.PORT || 5000;
app.use(cors());
app.listen(port, () => {
  console.log(`Server running on port ${port}`);
});

```
My package.json:
```json
{
  "name": "flamboyantknowndesigner",
  "version": "1.0.0",
  "description": "",
  "main": "server.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "node server.js"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "cors": "^2.8.5",
    "dotenv": "^16.0.3",
    "express": "^4.18.2",
    "mysql": "^2.18.1",
    "nodemon": "^2.0.20"
  }
}
```
My package-lock.json file:
```json
    "node_modules/dotenv": {
      "version": "16.0.3",
      "resolved": "https://registry.npmjs.org/dotenv/-/dotenv-16.0.3.tgz",
      "integrity": "sha512-7GO6HghkA5fYG9TYnNxi14/7K9f5occMlp3zXAuSxn7CKCxt9xbNWG7yF8hTCSUchlfWSe3uLmlPfigevRItzQ==",
      "engines": {
        "node": ">=12"
      }
    },
```
My Build Settings:
![image](https://github.com/vercel/community/assets/23269296/65e497eb-3d34-4c41-8347-88a8a82901bd)
Error:
![image](https://github.com/vercel/community/assets/23269296/ecfdb169-748a-43e8-b981-dc530b1cc96d)

### Steps to Reproduce

1) Please try deploying the project mentioned above, including all the dependencies and source code provided.

**Comment by amyegan** [comment, marked as answer]
Hi @GhouI! Your "Steps to Reproduce" say to try deploying the project mentioned above, but I don't see a link to a public repo. 😅  Can you share a link to a minimal reproduction, or your repo if it is public, so I can get a better look at the problem?

If you haven't already read it, I recommend our guide on using Express.js with Vercel to get an Express app working in a serverless environment. You can find the guide here: https://vercel.com/guides/using-express-with-vercel

**Comment by GhouI** [comment]
Thank you for the assistance.. I have looked at the guide and it seemed I was missing vercel.json file.
